ckanext/canada/plugins.py

```diff
@@ -102,6 +104,22 @@ def get_validators(self):
                 'canada_security_upload_presence':
                     validators.canada_security_upload_presence}
 
+    # IAuthenticator
+    def identify(self):
+        controller, action = p.toolkit.get_endpoint()
+        blueprint = '.'.join((controller, action))
+        restricted_blueprints = [
+            'canada.login',
+            'user.login',
+            'user.request_reset',
+            'canada.recover_username',
+            'canada.register',
+            'canada.action',
+            'api.action', # change if need to narrow down the scope
+        ]
+        if blueprint in restricted_blueprints and not helpers.registry_network_access():
+            return abort(403)
```

Comment:
might want to log failed access attempts so we can report on them

Reply:
Good idea. Do you think we need a separate log handler for "Registry Access"? As we will need to also log failed login attempts and all login sessions.

We could have a new log handler for `ckanext.canada.user_access` to go to its own log file `ckan_registry_access.log`.

Or should we just keep it in the normal logs and have Log Analytics handle all this stuff @wardi ?

Reply:
@JVickery-TBS I don't have a strong opinion either way. As long as the data can be pulled out of Log Analytics it shouldn't matter.

Reply:
@wardi and @JVickery-TBS let's implement logging as a separate feature for both registry network access and login access. It is one of the requirements given to us by imtd/security. I would prefer to implement it as a separate log handler so that it is easier for us to investigate any issues reported by helpdesk.